webapp/channels/src/components/file_preview_modal/image_preview.tsx

```diff
@@ -27,22 +72,157 @@ export default function ImagePreview({fileInfo, canDownloadFiles}: Props) {
         previewUrl = fileInfo.has_preview_image ? getFilePreviewUrl(fileInfo.id) : fileUrl;
     }
 
+    const computeScaleOptions = React.useCallback(() => {
+        const imgEl = imageRef.current;
+
+        if (imgEl) {
+            setBaseScale(imgEl.naturalWidth / imgEl.clientWidth);
+            setScaleOptions([
+                {
+                    key: 'auto',
+                    value: 1,
+                    label: formatMessage({id: 'file_preview_modal_image_zoom_actions.select.auto', defaultMessage: 'Automatic zoom'}),
+                },
+                {
+                    key: 'actual',
+                    value: Math.round((imgEl.naturalWidth / imgEl.clientWidth) * 1000) / 1000, // round to 3 decimal places to prevent a duplicate value conflict with 100%
+                    label: formatMessage({id: 'file_preview_modal_image_zoom_actions.select.actual_size', defaultMessage: 'Actual size'}),
+                },
+                ...[50, 75, 100, 125, 150, 200, 300, 400].map((level) => ({
+                    key: `${level}%`,
+                    value: Math.round((imgEl.naturalWidth / imgEl.clientWidth) * level) / 100,
+                    label: `${level}%`,
+                })),
+            ]);
+        }
+    }, [formatMessage]);
+
+    React.useEffect(() => {
+        const imgEl = imageRef.current;
+
+        const handleLoad = () => {
+            computeScaleOptions();
+        };
+
+        imgEl?.addEventListener('load', handleLoad);
+
+        return () => {
+            imgEl?.removeEventListener('load', handleLoad);
+        };
+    }, [imageRef, computeScaleOptions]);
+
+    React.useEffect(() => {
+        const onFullscreenChange = () => {
+            // delay for fullscreen to actually toggle on / off
+            setTimeout(computeScaleOptions, 100);
+        };
+
+        document.addEventListener('fullscreenchange', onFullscreenChange);
+
+        return () => {
+            document.removeEventListener('fullscreenchange', onFullscreenChange);
+        };
+    }, [computeScaleOptions]);
+
     if (!canDownloadFiles) {
         return <img src={previewUrl}/>;
     }
 
     return (
-        <a
-            className='image_preview'
-            href='#'
+        <TransformWrapper
+            onTransformed={(e) => setScale(e.state.scale)}
```

Comment:
Can you use `useCallback` for this since it's passed into `TransformWrapper` as a prop?